### PR TITLE
v1.13 Backports - background node sync improvements

### PR DESCRIPTION
**Pull request by marseel**
* [x] #32577 (@marseel)
 * [x] #32630 (@marseel)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 32577 32630
```

### CODE REVIEW

**Comment by marseel** [comment]
/test-backport-1.13

File: pkg/node/manager/manager.go [hunk 1]
```diff
@@ -316,17 +317,19 @@ func (m *Manager) backgroundSyncInterval() time.Duration {
...
+		startWaiting := syncTimer.After(syncInterval)
 		log.WithField("syncInterval", syncInterval.String()).Debug("Starting new iteration of background sync")
 		m.singleBackgroundLoop(syncInterval)
 		log.WithField("syncInterval", syncInterval.String()).Debug("Finished iteration of background sync")
 
 		select {
 		case <-m.closeChan:
 			return
-		default:
+		case <-startWaiting:
```
Comment (giorio94) [inline pkg/node/manager/manager.go:332]: Nit. The comment here has been lost during backporting.

```
// This handles cases when we didn't fetch nodes yet (e.g. on bootstrap)
// but also case when we have 1 node, in which case rate.Limiter doesn't
// throttle anything.
```

File: pkg/node/manager/manager.go
Comment (giorio94) [inline pkg/node/manager/manager.go:1]: Nit. The commit message still references v1.14, rather than v1.13.

**Review by giorio94 — APPROVED** [review]
/lgtm, thanks!

A couple of nits inline, given that I saw them. But feel free to ignore to avoid another CI round.

**Comment by marseel** [comment]
yea, I will skip that due to CI. Seems like Jenkins is having a tough time today.

**Review by qmonnet — APPROVED** [review]

**Pull request merged**